### PR TITLE
Add `stdlib.fetch.fetch()`, a way to fetch input files for build manifests.

### PR DESCRIPTION
**Pull request by Arignir**

### CODE REVIEW

File: stdlib/fetch.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+import hashlib
+import requests
+import ftplib
+from urllib.parse import urlparse
+
+
+def fetch():
+    """Fetch the input data contained in the versionized argument ``fetch``.
+
+    The versionized argument ``fetch`` must be an array of dictionnaries. Each entry in the array is a data to fetch.
```
Comment (doom) [inline stdlib/fetch.py]:
```suggestion
    The versionized argument ``fetch`` must be an array of dictionaries. Each entry in the array is a data to fetch.
```

File: stdlib/fetch.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    The dictionnary indicates how to retrieve the data and some optional parameters.
```
Comment (doom) [inline stdlib/fetch.py]:
```suggestion
    The dictionary indicates how to retrieve the data and some optional parameters.
```

File: stdlib/fetch.py [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+
+    If the dictionnary contains the key:
```
Comment (doom) [inline stdlib/fetch.py]:
```suggestion
    If the dictionary contains the key:
```

File: stdlib/fetch.py [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+
+        * ``url``: the function :py:func:`.fetch_url` is called with the remaining values as arguments.
+        * ``file``: the function :py:func:`.fetch_file` is called with the remaining values as arguments.
+
+    If the dictionary contains both the key ``url`` and ``file``, the behaviour of the function is undefined.
```
Comment (doom) [inline stdlib/fetch.py]: Are you sure this is the appropriate approach here ? We don't really benefit from undefined behaviour, so you might just wanna raise an exception instead to report the error

File: stdlib/fetch.py [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+                'url': 'https://example.com/hello.tar.gz',
+                'sha256': '2cf24dba5fb0a30e26e83b2ac5b9e29e1b161e5c1fa7425e73043362938b9824'
+            }, {
+                'url': 'https://example.com/world.tar.gz',
+            }, {
+                'file': './hello_world'
+            },
+        ]
+
+    With the above example, ``fetch`` will do those calls, respectively: ::
```
Comment (doom) [inline stdlib/fetch.py]:
```suggestion
    With the above example, ``fetch`` will perform these calls, respectively: ::
```

File: stdlib/fetch.py [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+        if 'url' in input:
+            fetch_url(**input)
+        elif 'file' in input:
+            fetch_file(**input)
+        else:
+            print("Bouuuh")
+
+
+def fetch_file(file: str):
+    """Copy a file or directory tothe current build cache.
```
Comment (doom) [inline stdlib/fetch.py]:
```suggestion
    """Copy a file or directory to the current build cache.
```

File: stdlib/fetch.py [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+
+    :note: The path must be relative to the build manifest. If an aboslute path is given, a ``RuntimError`` is raised.
```
Comment (doom) [inline stdlib/fetch.py]:
```suggestion
    :note: The path must be relative to the build manifest. If an absolute path is given, a ``RuntimeError`` is raised.
```

**Review by doom — CHANGES_REQUESTED** [review]

File: stdlib/fetch.py [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+        stdlib.log.wlog(f"No sha256 to ensure the integrity of {url}")
+
+    if os.path.exists(install_path) and _check_sha256(install_path, sha256):
+        stdlib.log.slog(f"Cache hit for {url}")
+        shutil.copy2(
+            install_path,
+            build_path
+        )
+    else:
+        stdlib.log.ilog(f"Cache miss for {url}, fetching now...")
```
Comment (Pedrochu) [inline stdlib/fetch.py:135]:
```suggestion
        stdlib.log.ilog(f"Cache missing for {url}, fetching now...")
```

**Review by Pedrochu — CHANGES_REQUESTED** [review]

File: stdlib/fetch.py (hunk 8 above)
Comment (Arignir) [reply stdlib/fetch.py:135]: I'll keep the previous one, it's a common expression

**Review by doom — APPROVED** [review]

**Pull request merged**